core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -369,11 +383,23 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
 
         CompletableFuture<Void> allFutures = CompletableFuture.allOf(
             futuresMap.values().toArray(new CompletableFuture[0]));
-        return allFutures.thenApply(v -> {
+        return allFutures.handle((results, exception) -> {
             Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData> result = new HashMap<>();
-            futuresMap.forEach((topicIdPartition, future) -> result.put(topicIdPartition, new ShareAcknowledgeResponseData.PartitionData()
-                    .setPartitionIndex(topicIdPartition.partition())
-                    .setErrorCode(future.join().code())));
+            futuresMap.forEach((topicIdPartition, future) -> {
+                try {
+                    Errors error = future.join();
+                    result.put(topicIdPartition, new ShareAcknowledgeResponseData.PartitionData()
+                            .setPartitionIndex(topicIdPartition.partition())
+                            .setErrorCode(error.code()));
```

Comment:
I think there are situations (such as UNKNOWN_TOPIC_OR_PARTITION) where the future completes with a code other than Errors.NONE, but does not complete exceptionally. I suggest that the default exception string for the error should be used in this situation.

Reply:
Done.